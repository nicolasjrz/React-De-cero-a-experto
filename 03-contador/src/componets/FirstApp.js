import React from "react";

export const FirstApp = () => {
  return (
    <>
      <h1>Hellow word app</h1>
      <p>jeje holis</p>
    </>
  );
};
