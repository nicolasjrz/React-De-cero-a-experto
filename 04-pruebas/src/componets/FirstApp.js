import React from "react";
import PropTypes from "prop-types";
const FirstApp = ({ title, subtitle, parrafo }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>

      <p>{subtitle}</p>
      <p>{parrafo}</p>
    </>
  );
};

export default FirstApp;

FirstApp.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
};

FirstApp.defaultProps = {
  //title: "Prueba de react",
  subtitle: "no hay subtitulo",
  parrafo: "no hay parrafo",
};
