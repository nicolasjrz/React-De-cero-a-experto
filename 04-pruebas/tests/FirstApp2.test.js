import { render, screen } from "@testing-library/react";
import FirstApp from "../src/componets/FirstApp";
describe("Prueba en <>FirstApp</>", () => {
  const title = "hola soy goku";
  const subtitle = "hola soy un subtitulo";
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe mostar el mensaje "hola soy goku"', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe mostrar el tittulo en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe de mostgar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);

    expect(screen.getAllByText(subtitle).length).toBe(1);
  });
});
