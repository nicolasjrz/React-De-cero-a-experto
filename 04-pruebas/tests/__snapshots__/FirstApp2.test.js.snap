// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <>FirstApp</> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola soy goku
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay parrafo
  </p>
</div>
`;
